helixbass/tsc-rust
# Flag unknown property on object literal

In this PR:
- flag a type error when an unknown property exists on an object literal that's initializing a variable of a certain interface type

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
interface Foo {
  bar: number
}

const a: Foo = {
  bar: 1,
  baz: 2,
}
```
then you should see a diagnostic debug-logged with chained `message_text` `"Type '{ baz: number; bar: number; }' is not assignable to type 'Foo'."` (not sure why `baz` is printed before `bar`, probably not sorting something correctly or something) + `"Object literal may only specify known properties, and 'baz' does not exist in type 'Foo'."`

Based on `type-check-object-literal`
